src/server/qgsserver.cpp

```diff
@@ -415,7 +415,7 @@ void QgsServer::handleRequest( QgsServerRequest &request, QgsServerResponse &res
         else
         {
           throw QgsOgcServiceException( QStringLiteral( "Service configuration error" ),
-                                        QStringLiteral( "Service unknown or unsupported" ) );
+                                        QStringLiteral( "Service unknown or unsupported. Current supported services (case-sensitive): WMS WFS WCS WMTS SampleService, or use a WFS3 (OGC API Features) endpoint" ) );
```

Comment:
You can probably dynamically get the list of available service and APIs directly from the server, it would be safer in case the user added custom services or APIs through plugins or in case we add new core ones in the future.

Reply:
@elpaso yes, I already found: 
server.sServiceRegistry.mServiceVersions
server.sServiceRegistry.mApiVersions
but before putting to much effort in it I first thought to wait for you to see IF you want all this or not :-) I'll happy to try.

Reply:
I think that listing the supported services and APIs is a good idea.
CC @pblottiere @rldhont what do you think?

Reply:
> I think that listing the supported services and APIs is a good idea.

+1

Thanks for taking a look @rduivenvoorde :)

Reply:
@elpaso @pblottiere  why is QGIS so strict about the service parameter case?
'WMS' is ok, but 'wms' fails with (now) error above?
Should we not just cast to uppercase and check for 'WMS'? 
(I know the specs state: keys are case-INsensitive, and value are case-sensitive), but in this case both could/should work (like in all other WMS's I wordt with?)

Reply:
> I think that listing the supported services and APIs is a good idea.

+1

Reply:
> why is QGIS so strict about the service parameter case?
'WMS' is ok, but 'wms' fails with (now) error above?

It's a good question actually :). It should be checked but I don't think it's an OGC curb...